### DISCUSSION TITLE
PxRigidDynamic with Skeleton

### DISCUSSION THREAD

**Discussion by rossetti73** [NVIDIA-Omniverse/PhysX#194] [Q&A]
Hi ,
Is possible something similar Skeleton on object PxRigidDynamic?

i check many example  but all create many object  PxRigidDynamic and after modify setKinematicTarget but i want only one mesh with many target to simulate skeleton .

is possible thanks ?

**Comment by preist-nvidia** [comment]
Hi @rossetti73,

If I understand you correctly, you want to animate a triangle mesh that then collides with the environment? We only support this on CPU and only for static colliders at the moment, see: https://github.com/NVIDIA-Omniverse/PhysX/blob/main/physx/snippets/snippetdeformablemesh/SnippetDeformableMesh.cpp

~~For dynamics we have a GPU-accelerated feature planned, but it is not released yet.~~

Philipp

**Reply by rossetti73** [reply]
Hi thanks for answer .
so until now not possible collision with Particle System  like a Cloth is only on GPU?
when do you think release new features ?

**Reply by preist-nvidia** [reply]
I am sorry, I am mistaken, the feature is already released. The kinematic softbody snippet does exactly what you need: https://github.com/NVIDIA-Omniverse/PhysX/tree/main/physx/snippets/snippetkinematicsoftbody

But yes, this feature is GPU only and will remain so.

**Reply by rossetti73** [reply]
thanks for replay  So no way to use RigidDynamic? because use softbody  not possible use all mesh need prepare and modify to made softbody 

i try update RigidDynamic like SnippetDeformableMesh.cpp  update position in render but collision with cloth not update !!
Only way i find to update RigidDynamic is complete remove and remade with new position in this way work but is very slow!!!
not way to update position of RigidDynamic in GPU?

thanks for any help

**Reply by preist-nvidia** [reply]
Why is creating a softbody an issue? That's your best option to make this work with cloth.

Yes, the deformable mesh feature is only for rigid statics, and will also not work with GPU features like particle cloth.

**Reply by rossetti73** [reply]
About Create soft body  i have this manikin with also animation 

<img width="432" alt="1695218701558" src="https://github.com/NVIDIA-Omniverse/PhysX/assets/72785510/9714861e-fde4-4e62-a121-26940cd320fc">

This is many mesh not connected .
So when i cook mesh  for softbody  not work i need before simplify mesh also connect.
If in blender connect my manikin and simplify still many position and to cook it  many time !!


About collision with   RigidDynamic    work in GPU  and very easy to cook and faster !!
<img width="409" alt="1695219441768" src="https://github.com/NVIDIA-Omniverse/PhysX/assets/72785510/2cccb1ca-a411-495e-a699-733efdcaad77">

i cook with 
params.buildGPUData = true;  //rigid body collide in GPU

but my problem i can't update position,  if try update lost collision, because i think not update position in gpu.

but if delete actor in scene, and cook mesh in new position, and put again in scene  actor, work  but for animation is very slow!!

for this reason i ask is possible update position of  RigidDynamic   in GPU .  (GPUDATA)

If now no, do you think in the future add it ?

In flex is possible i do it :)

Thanks .

**Reply by preist-nvidia** [reply]
Yes, reinserting into the scene will be very slow - that's why the best option is the softbody approach.

An alternative is to approximate the collision shape of the skeleton with primitive shapes, or convex hulls of the mesh components, and then drive the visual mesh from the skeleton. But that will be more work than fusing the mesh for kinematic softbody I think.
